### PR TITLE
draft: ffmpeg5 split

### PR DESCRIPTION
**Pull request by zlice**
[ci skip]

#### Testing the changes

- I tested the changes in this PR: **YES**

~~I have conflicting changes from [intel packages](https://github.com/void-linux/void-packages/pull/41132) and don't want to mess with my system. If it makes sense I'll move these changes here like I originally intended. ffmpeg5 makes Intel vaapi easier and only rely on oneVPL iirc.~~

I do not use 80%+ of these packages. I may be able to run Blender, and I use shotcut.

#### New package

- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing

- I built this PR locally for my native architecture, **x86_64**

#### Discussion

Continuation of https://github.com/void-linux/void-packages/pull/36315 - thx tibequadorian :)

It was suggested to make `ffmpeg` version 5 and 'copy' (create a new?) `ffmpeg4` package, and that seems to work well.

As stated, I haven't and probably won't test most of these packages, hoping for others to voice in on what's working. 

A few months back (and now) afaict everything has been compiling with ffmpeg5 except for the few problem packages still using ffmpeg4 (below). Can add to the ffmpeg4 list as a incompatible package is found.

`libavresample` is deprecated, but marked as a dependency when trying to build other packages and install `ffmpeg`(5). I think the right way to handle this is make a meta package and the add to removals?

[gstreamer 1.22.1 updates](https://github.com/void-linux/void-packages/pull/42861) could go before this

##### Commit naming / misc

~~`ffpmeg4` isn't "new", not sure what the commit messages should be really for the 4 and 5 split. Additionally xlint complains about `ffmpeg4` revision being 3 and not 1 because it's a "new" package, I assume that's right but not sure.~~

~~Not sure how commits should be separated. ffmpeg4/5 could be split and all the rev bumps included (otherwise it's over 100 commits). opencv was updated to 4.7.0 for ffmpeg5 so I figured that should be a separate commit?~~

I wouldn't mind moving the [intel packages](https://github.com/void-linux/void-packages/pull/41132) into here under another commit. That just got a potential fix that may allow merging but I imagine moving to ffmpeg5 will change and even remove some of those packages. (I won't really test these changes without Intel vaapi either way.)

Maintainer for previous ffmpeg(4) was orphaned, I can put my name in there if that matters, though I don't exactly see myself keeping up on ffmpeg.

##### STILL FFMPEG4

These are a mix of emulators, games and/or older packages.

- attract (multiple patches in master)
- audacity  (3.2.x should work but void is missing some libs? vst3sdk?)
- ccextractor (master patches? no release)
- vice
- mlt
- synfig (mlt)
- tvheadend
- ssr
- spek-alternative
- ppsspp
- olive
- moc
- mediastreamer
- libextractor
- idjc
- hedgewars
- ~~handbrake~~
- ffmpegthumbnailer
- ffmpegthumbs
- deadbeef
- omxplayer
- qtav (fork has ffmpeg 5 patch - https://github.com/cmguo/QtAV)
- ~~qt5-webengine~~ (need patch and version bump. ~~may work? if not gentoo has a patch? https://bugs.gentoo.org/831487~~)
- `vlc` (vlc4 will support ffmpeg5, there's a patch to build 3.x with ffmpeg5 but it will lack vaapi)
- `osg` (OpenSceneGraph plus `xine-lib`)
- `ytmdl` ? not sure but it uses `python3-ffmpeg-python` which being 6-8yr old I imagine is ffmpeg4

##### HAVE NOT BUILT YET (big-uns)

- omxplayer (rpi only?)
- kodi
- electron19
- ~~qt5-webengine~~ - some patches and version update
- ~~digikam~~ (qt5-webengine)
- ~~chromium~~ - ~~some patches but vaapi doesnt seem to be working~~ (use special flags)

##### UPDATED (for ffmpeg5)

- unpaper: update to 7.0.0.
- guvcview: update to 2.0.8.

- qt5: update to 5.15.8
- qt5-webengine: update to 5.18.13
- shiboken2: revbump for qt5.15.8 (patch)
- python3-pyside2: update to 5.15.8 (patch)

### CODE REVIEW

**Comment by paper42** [comment]
> fpmeg4 isn't "new", not sure what the commit messages should be really for the 4 and 5 split

It is a new package

> Not sure how commits should be separated. ffmpeg4/5 could be split and all the rev bumps included (otherwise it's over 100 commits). opencv was updated to 4.7.0 for ffmpeg5 so I figured that should be a separate commit?

as mentioned in CONTRIBUTING.md, use one commit per package

> Testing the changes
>
>    I tested the changes in this PR: NO
>
> I have conflicting changes from https://github.com/void-linux/void-packages/pull/41132 and don't want to mess with my system.

I don't think it's a good idea to blindly do a complex update like this and not even test it.

File: srcpkgs/mesa/template [hunk 1]
```diff
@@ -1,12 +1,13 @@
 # Template file for 'mesa'
 pkgname=mesa
-version=22.3.5
+version=23.0.0
```
Comment (paper42) [inline srcpkgs/mesa/template:3]: this looks unrelated

File: srcpkgs/fluxbox/template [hunk 2]
```diff
@@ -1,29 +1,56 @@
 # Template file for 'fluxbox'
 pkgname=fluxbox
-version=1.3.7
-revision=4
+version=1.3.8
+#nostrip=nostrip
```
Comment (paper42) [inline srcpkgs/fluxbox/template]: also unrelated?

File: srcpkgs/ffmpeg4/ffmpeg4 [hunk 3]
```diff
@@ -0,0 +1 @@
+ffmpeg4
```
Comment (paper42) [inline srcpkgs/ffmpeg4/ffmpeg4:1]: random useless symlink

File: srcpkgs/ffmpeg4/template [hunk 4]
```diff
@@ -0,0 +1,254 @@
+# Template file for 'ffmpeg4'
+# audacity also needs to be bumped when a new ffmpeg version bumps libavformat's soname!
+pkgname=ffmpeg4
+version=4.4.3
+revision=3
```
Comment (paper42) [inline srcpkgs/ffmpeg4/template]: 1

File: srcpkgs/ffmpeg/template [hunk 5]
```diff
@@ -198,9 +178,49 @@ ffmpeg-devel_package() {
 		libavfilter>=${version}_${revision}
 		libpostproc>=${version}_${revision}
 		libswscale>=${version}_${revision}
-		libswresample>=${version}_${revision}
-		libavresample>=${version}_${revision}"
+		libswresample>=${version}_${revision}"
 	short_desc+=" - development files"
+	alternatives="
```
Comment (paper42) [inline srcpkgs/ffmpeg/template]: I think you should just remove the manpages and let -devel packages conflict. This will only cause issues.

File: srcpkgs/ffmpeg/template [hunk 6]
```diff
@@ -98,39 +76,47 @@ do_configure() {
 		_args+=" --extra-libs=-latomic"
 	fi
 
-	./configure --prefix=/usr --disable-debug --enable-gpl --enable-gnutls \
+	./configure \
+		--prefix='/usr' \
+		--extra-cflags='-I/opt/cuda/include' \
+		--extra-ldflags='-L/opt/cuda/lib64' \
```
Comment (paper42) [inline srcpkgs/ffmpeg/template]: what's the purpose of this?

File: srcpkgs/fluxbox/template (hunk 2 above)
Comment (zlice) [reply srcpkgs/fluxbox/template]: ya accidentally added that

File: srcpkgs/mesa/template (hunk 1 above)
Comment (zlice) [reply srcpkgs/mesa/template:3]: accidental add

**Comment by paper42** [comment]
due to the large amount of random issues I found, I am not confident that the parts of this PR that I didn't check (like the huge lists of dependencies) don't contain similar random mistakes. Even if everything gets fixed, I don't think we should merge this without carefully checking every single line which is probably way harder than just doing the update again.

File: srcpkgs/ffmpeg/template (hunk 6 above)
Comment (zlice) [reply srcpkgs/ffmpeg/template]: switched to the branch i was using instead of another, not intended again sorry

**Comment by sgn** [comment]
I'm advocate for not using major version of package as suffix for libraries' package name, instead, I think those libaries should use its SOVERSION for pkgname, like Debian.
For example, the new `libswscale4` should be named `libswscale5`, the one splited from `ffmpeg5` will be named `libswscale6`.

**Comment by zlice** [comment]
K everything should be building except the few outliers I added to OP.

Also, should we change this message? `# audacity also needs to be bumped when a new ffmpeg version bumps libavformat's soname!` seems like anything with `ffmpeg-devel` and I'm not 100% convinced that covers it (ytmdl doesn't have 'devel' but ffmpeg in 'depends')

**Comment by zlice** [comment]
fwiw, all ffmpeg5 packages seem to compile with a few extra patches for ffmpeg6

not sure if skipping 5 to 6 is fine (6 has all the av1 codec stuff i personally want)

**Comment by zlice** [comment]
installed ffmpeg6 last night https://github.com/zlice/void-packages/tree/ffmpeg6-split

chromium is the only exception so far i think

**Comment by zlice** [comment]
Have been running ffmpeg6 fine for a bit now, though on top of my `gstreamer` and `intel packages` PRs mentioned. Want to double check commits on the 6 branch even without gstreamer/intel changes.

This does touch a lot and is relying on other big changes. What should next steps be? (will probably make another PR with the 6 branch at some point)

**Comment by zlice** [comment]
Closing this in favor of https://github.com/void-linux/void-packages/pull/43761 I have been running ffmpeg6 for a month now.
